ironfish/src/primitives/target.test.ts

```diff
@@ -64,81 +65,86 @@ describe('Target', () => {
-      const newTarget = Target.calculateTarget(time, now, target)
-
-      expect(newDifficulty).toBeLessThan(difficulty)
-      expect(BigInt(newDifficulty) + diffInDifficulty).toEqual(difficulty)
-
-      expect(newTarget.asBigInt()).toBeGreaterThan(target.asBigInt())
-    }
-
-    // for any time 30-39 seconds after the last block, difficulty should decrease by previous block's difficulty / BigInt(2048) * 2
-    for (let i = 1; i < 10; i++) {
-      const time = new Date(now.getTime() + 30000 + i * 1000)
-
-      const difficulty = BigInt(231072)
-      const target = Target.fromDifficulty(difficulty)
+      const bucketFromParent =
+        TARGET_BLOCK_TIME_IN_SECONDS / FREEZE_TIME_IN_SECONDS -
+        Math.round(i / FREEZE_TIME_IN_SECONDS)
 
-      const diffInDifficulty = (difficulty / BigInt(2048)) * BigInt(2)
+      const diffInDifficulty = (difficulty / BigInt(2048)) * BigInt(bucketFromParent)
 
       const newDifficulty = Target.calculateDifficulty(time, now, difficulty)
       const newTarget = Target.calculateTarget(time, now, target)
 
-      expect(newDifficulty).toBeLessThan(difficulty)
-      expect(BigInt(newDifficulty) + diffInDifficulty).toEqual(difficulty)
+      expect(newDifficulty).toBeGreaterThan(difficulty)
+      expect(BigInt(difficulty) + diffInDifficulty).toEqual(newDifficulty)
 
-      expect(newTarget.asBigInt()).toBeGreaterThan(target.asBigInt())
+      expect(newTarget.targetValue).toBeLessThan(target.targetValue)
     }
   })
 
-  it('can decrease target (which increases difficulty) if a block is trying to come in too early (1-10 seconds)', () => {
+  it('keeps difficulty/target of parent block header if time differnece is between 55 and 65 seconds', () => {
     const now = new Date()
-    for (let i = 1; i < 10; i++) {
+    for (let i = 55; i < 65; i++) {
       const time = new Date(now.getTime() + i * 1000)
 
       const difficulty = BigInt(231072)
       const target = Target.fromDifficulty(difficulty)
 
-      const diffInDifficulty = difficulty / BigInt(2048)
-
       const newDifficulty = Target.calculateDifficulty(time, now, difficulty)
       const newTarget = Target.calculateTarget(time, now, target)
 
-      expect(newDifficulty).toBeGreaterThan(difficulty)
-      expect(BigInt(difficulty) + diffInDifficulty).toEqual(newDifficulty)
+      const diffInDifficulty = BigInt(newDifficulty) - difficulty
 
-      expect(newTarget.targetValue).toBeLessThan(target.targetValue)
+      expect(diffInDifficulty).toEqual(BigInt(0))
+      expect(newTarget.targetValue).toEqual(target.targetValue)
     }
   })
 
-  it('keeps difficulty/target of parent block header if time differnece is between 10 and 20 seconds', () => {
+  it('decreases difficulty if a new block is coming in after the target range time', () => {
     const now = new Date()
-    for (let i = 10; i < 20; i++) {
+
+    /**
+     * if new block comes after target block mining time + half bucket time, then difficulty is adjust as:
```

Comment:
`half bucket time` ? 

I know what you mean, but our code is using static variables that don't mention bucket. we have something like `FREEZE_TIME_IN_SECONDS` which is very confusing. maybe we can rename `FREEZE_TIME_IN_SECONDS` to something less confusing and reference it here? or define bucket somewhere?